packages/core/src/utils/Utils.ts

```diff
@@ -1360,3 +1360,76 @@ export class Utils {
   }
 
 }
+
+/**
+ * @internal
+ */
+export interface TryModuleErrorOptions extends ErrorOptions {
+  cause: NodeJS.ErrnoException;
+}
+
+/**
+ * This error is a noop flavor for `ERR_MODULE_NOT_FOUND` error with required `cause` option.
+ *
+ * @internal
+ */
+
+export class TryModuleError extends Error {
```

Comment:
```suggestion
 */
export class TryModuleError extends Error {
```

also maybe think of a better name? but i guess we can keep this too, it just feels weird to have the `Try` prefix in an error name.

and move this to the https://github.com/mikro-orm/mikro-orm/blob/678119475cee860dd4684c9f4d903a3d4028b8a3/packages/core/src/errors.ts file

---

btw why is it internal, do we always catch it internally or can it bubble to the user? if it can, i wouldn't mark it as internal

Reply:
> btw why is it internal, do we always catch it internally or can it bubble to the user? if it can, i wouldn't mark it as internal

It is for internal use only. We will catch it and use its data (that's what is the `specifier` field for) to display a message asking for ts-node/jiti/tsx installation. Unlike Node.js' errors it can be tested with `instanceof` and type casting is not needed anymore.

I think `tryModule` and `requireDefault` should also be inaccessible from `@mikro-orm/core` package, since they supposed to use internally.

Reply:
> I think tryModule and requireDefault should also be inaccessible from @mikro-orm/core package, since they supposed to use internally.

Makes sense, as long as it won't be part of the barrel file, it shouldn't be exported. I guess we should rename them to `exports.ts` or similar, I don't want to use them internally (it can cause issues with cycles), they should be there currently only for this use case.

Reply:
> but you'll have to adjust the barrel file, I think it now reexports this whole file.

Which one? The Utils? Yup, You can see how I access my utils in the tests.

By the way, is the `errors.ts` also a barrel file? Some of the errors must be part of the public API surface, so you probably re-export them too, and I want `TryModuleError` class (or whatever the name will be at the end) to be inaccessible outside of `@mikro-orm/core`, since it's for internal use only.

Reply:
> Which one? 

https://github.com/mikro-orm/mikro-orm/blob/8f4790f01b0704cb18d31ff89686ec2eb80b6537/packages/core/src/utils/index.ts#L5

> By the way, is the errors.ts also a barrel file?

Barrel files reexport things, so in that sense, nope, its not. But we reexported all its contents as part of the upper barrel file.

https://github.com/mikro-orm/mikro-orm/blob/18a425c091af0535d6b43dfb797edade58822fd2/packages/core/src/index.ts#L16

Maybe move all of this into the `loaders.ts` file and don't export it at all?

Reply:
Sounds good, I'll move all these new utilities into loaders.ts, and if you'll ever need tryModule or requireDefault, you'll just change the code.

Reply:
Moved this error to loader.ts. I don't think the name matters quite a lot, but I can rename it when I figure out how to test this error's appearance (I think it can be done by mocks if I throw NodeJS.ErrnoError instead of returning the exports?), sure.